dogpile/cache/api.py

```diff
@@ -8,6 +9,10 @@
 from typing import Optional
 from typing import Sequence
 from typing import Union
+if sys.version_info >= (3, 11):
```

Comment:
can this conditional be moved to dogpile/util/compat.py , or a new module dogpile/util/typing.py?  then we import as "from ..util.typing import Self"  thanks

Reply:
I was thinking about doing this as well